src/toil/test/provisioners/aws/awsProvisionerTest.py

```diff
@@ -191,28 +200,27 @@ def _test(self, preemptibleJobs=False):
 
         assert len(self.cluster._getRoleNames()) == 1
 
-        from boto.exception import EC2ResponseError
         volumeID = self.getRootVolID()
         self.cluster.destroyCluster()
+        boto3_ec2: EC2Client = self.aws.client(region=self.region, service_name="ec2")
+        volume_filter: FilterTypeDef = {"Name": "volume-id", "Values": [volumeID]}
+        volumes: Optional[List[VolumeTypeDef]] = None
         for attempt in range(6):
             # https://github.com/BD2KGenomics/toil/issues/1567
             # retry this for up to 1 minute until the volume disappears
-            try:
-                self.boto2_ec2.get_all_volumes(volume_ids=[volumeID])
-                time.sleep(10)
-            except EC2ResponseError as e:
-                if e.status == 400 and 'InvalidVolume.NotFound' in e.code:
-                    break
-                else:
-                    raise
-        else:
+            volumes = boto3_ec2.describe_volumes(Filters=[volume_filter])["Volumes"]
+            if len(volumes) == 0:
+                # None are left, so they have been properly deleted
+                break
+            time.sleep(10)
+        if volumes is None or len(volumes) > 0:
             self.fail('Volume with ID %s was not cleaned up properly' % volumeID)
 
         assert len(self.cluster._getRoleNames()) == 0
 
 
 @integrative
-@needs_mesos
+# @needs_mesos
```

Comment:
Doesn't this need the Mesos extra? It can't work on Python too new to have PyMesos, right?